metrics/resource_view.go

```diff
@@ -315,21 +315,49 @@ func resourceToKey(r *resource.Resource) string {
 	if r == nil {
 		return ""
 	}
+
+	// If there are no labels, we have just the Type.
+	if len(r.Labels) == 0 {
+		return r.Type
+	}
+
 	var s strings.Builder
+	writeKV := func(key, value string) {
+		// We use byte values 1 and 2 to avoid colliding with valid resource labels
+		// and to make unpacking easy
+		s.WriteByte('\x01')
+		s.WriteString(key)
+		s.WriteByte('\x02')
+		s.WriteString(value)
+	}
```

Comment:
not sure if at this point `func writeKV(Builder, string, string)` isn't better, or definitely more readable.
Shouldn't it save an allocation, since otherwise it'll dynamically create the lambda to bind it to  `s`?

Reply:
The lambda doesn't escape apparently. Moving it out works too though.